src/ApiPort.VisualStudio.Common/Analyze/ProjectAnalyzer.cs

```diff
@@ -117,5 +125,25 @@ private IEnumerable<string> GetAllAssembliesInDirectory(string directory)
                 return Enumerable.Empty<string>();
             }
         }
+
+        private IEnumerable<string> GetPackageReferences(Project project)
+        {
+            var componentModel = (IComponentModel)Microsoft.VisualStudio.Shell.ServiceProvider.GlobalProvider.GetService(typeof(SComponentModel));
+            IVsPackageInstallerServices installerServices = componentModel.GetService<IVsPackageInstallerServices>();
+            if (installerServices == null)
+            {
+                yield break;
+            }
+
+            var installedPackages = installerServices.GetInstalledPackages(project);
+
+            foreach (var packageMetadata in installedPackages)
+            {
+                if (!NuGetPackageInfo.IsImplicitlyReferencedPackage(packageMetadata.Id))
+                {
+                    yield return packageMetadata.Id;
```

Comment:
Much of the code in this project passes around `IEnumerable<string>` parameters, often two (one for assemblies, one for packages), making the ordering critical and the user's responsibility. If these collections were of some non-primitive type, the compiler could help, and the API could be less awkward in places (e.g. `IAnalysisEngine.GetNugetPackagesInfoFromAssembly` could become an overload of `.GetNugetPackagesInfo`.